### PR TITLE
Finalize the project

### PR DESCRIPTION
**Pull request by M-Anwar-Hussaini**
# Brief summery of the project:
- [x] Fetch data from API to show details, using redux middleware.
- [x] Follow the design from Behence template.
- [x] Creating the store from `redux/toolkit` for state management.
- [x] Home page to list general information for every city/province.
- [x] Detail page to show more details to the specific item.
- [x] Provide [video](https://drive.google.com/file/d/1ASjw86RXEpW6nJ7wqkIjQoIDJClAcWtf/view?usp=sharing) ducumentaion.
- [x] No linters error.
- [x] Perform tests for components using `RTL` and `jest`

### CODE REVIEW

File: src/pages/Home.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    dispatch(setHeader('Afghanistan'));
+  }, [dispatch]);
+
+  return (
+    <>
+      <Hero />
+      <ProvinceList />
+    </>
+  );
+}
```
Comment (DammyShittu) [inline src/pages/Home.jsx:19]: - [ ] On the home page you failed to add a search/filter feature. Kindly add this. Users should be filter elements on the home page.

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @M-Anwar-Hussaini,

## Highlights
✔️ Design and pattern
✔️ Unique routes for items

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by M-Anwar-Hussaini** [comment]
- [x] Search functionality added:
- image
- [x] Kindly check this [commit](https://github.com/M-Anwar-Hussaini/Weather-Forecast/pull/1/commits/3cae6646d8407d7ee51e99882397f88a5803760f) for the changes I brought.

File: src/components/ProvinceList.jsx [hunk 2]
```diff
@@ -12,34 +12,48 @@ const linkStyle = {
 };
 
 export default function ProvinceList() {
+  const [result, setResult] = useState('');
   const { provinces, isLoading } = useSelector((store) => store.provinces);
   const dispatch = useDispatch();
+
+  const filtered = provinces.filter((el) => el.name.toLowerCase().includes(result));
+
```
Comment (Gambit142) [inline src/components/ProvinceList.jsx]: - [OPTIONAL] I noticed that the search functionality only works when the user's input is in lowercase. When it is in uppercase, it doesn't work. I would advise you to convert the user's input to lowercase even if the user is typing in uppercase. 👍 😎 

#### Hint:

```JavaScript
const filtered = provinces.filter((el) => el.name.toLowerCase().includes(result.toLowerCase()));
```

**Review by Gambit142 — APPROVED** [review]
Hi @M-Anwar-Hussaini 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- All linters checks passed ✔️ 
- Created Home Page ✔️ 
- Created Details Page ✔️ 
- Implemented search functionality ✔️ 

### Optional Suggestion

#### You used `GitHub` flow instead of `Gitflow` for this project. You ought to submit a link from a feature branch to the dev branch as the base branch and not from the dev branch to the main branch as the base branch. 👍 

#### Kindly check the comment under this review tagged [OPTIONAL]

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/ProvinceList.jsx (hunk 2 above)
Comment (M-Anwar-Hussaini) [reply src/components/ProvinceList.jsx]: Hi @Gambit142 .
Thanks for your suggestion. I actually, noticed that issue and applied it. I guess, I might have pushed it after your review. Here is the [commit](https://github.com/M-Anwar-Hussaini/Weather-Forecast/pull/1/commits/6976282d813a2dda5bc825446a4e5f1a0a33596b).

**Comment by Gambit142** [comment]
Hi @M-Anwar-Hussaini, you are welcome. Great job 👍

**Pull request merged**